include/gemmi/topo.hpp

```diff
@@ -44,7 +44,21 @@ struct Topo {
       return calculate_dihedral(atoms[0]->pos, atoms[1]->pos,
                                 atoms[2]->pos, atoms[3]->pos);
     }
-    double calculate_z() const { return angle_z(calculate(), *restr); }
+    double calculate_z() const {
+      const double val = deg(calculate());
+      double closest = restr->value;
+      double min_diff = angle_abs_diff(closest, val);
+      double diff, tmp;
+      for (int i = 1; i < restr->period; ++i) {
+        tmp = restr->value + 360. * i / restr->period;
+        diff = angle_abs_diff(tmp, val);
```

Comment:
Unfortunately I'm unfamiliar with this piece of gemmi, so my comment is about code style only. 

It would be slightly better to avoid uninitialized variables, e.g.: 
```suggestion
      for (int i = 1; i < restr->period; ++i) {
        double tmp = restr->value + 360. * i / restr->period;
        double diff = angle_abs_diff(tmp, val);
```